common/src/main/java/com/wynntils/models/territories/GuildAttackTimerModel.java

```diff
@@ -13,30 +13,51 @@
 import com.wynntils.handlers.chat.type.RecipientType;
 import com.wynntils.handlers.scoreboard.ScoreboardPart;
 import com.wynntils.handlers.scoreboard.ScoreboardSegment;
+import com.wynntils.mc.event.TickEvent;
 import com.wynntils.models.marker.MarkerModel;
 import com.wynntils.models.territories.event.GuildWarQueuedEvent;
 import com.wynntils.models.territories.markers.GuildAttackMarkerProvider;
+import com.wynntils.models.territories.profile.TerritoryProfile;
 import com.wynntils.models.territories.type.GuildResourceValues;
-import com.wynntils.utils.type.Pair;
-import com.wynntils.utils.type.TimedSet;
+import com.wynntils.utils.mc.McUtils;
 import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
 import java.util.Optional;
-import java.util.concurrent.TimeUnit;
+import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Stream;
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 
+/**
+ * A note about territory attack timers:
+ * - Chat messages are sent on attack, and on certain time intervals
+ * - Scoreboard is always up to date, but can cut off vertically and horizontally
+ * (this means it can cut off long territory names and full lines, if there are too many attacks)
+ * <p>
+ * We use a hybrid approach:
+ * - Use chat messages when we can, as they are more reliable
+ * - Use scoreboard messages when we don't have a chat message
+ * - If a chat message is received for a territory, we stop using the scoreboard for that territory
+ * - We also take territory defenses from chat messages
+ */
 public final class GuildAttackTimerModel extends Model {
     private static final Pattern GUILD_ATTACK_PATTERN = Pattern.compile("§b- (.+):(.+) §3(.+)");
     private static final Pattern GUILD_DEFENSE_CHAT_PATTERN = Pattern.compile("§3.+§b (.+) defense is (.+)");
+    private static final Pattern WAR_MESSAGE_PATTERN = Pattern.compile(
+            "§3\\[WAR\\]§c The war for (?<guild>.+) will start in (?<remaining>.+) (?<type>minutes|seconds)\\.");
```

Comment:
Minor nitpick but the guild group should be called territory

Reply:
Thanks!